lua/pckr/loader/keys.lua

```diff
@@ -14,21 +14,19 @@ return function(mode, key, rhs, opts)
     opts.silent = true
   end
 
-  rhs = rhs
-    or function()
+  --- @param loader fun()
+  return function(loader)
+    local rhs_func = function()
+      -- always delete the mapping immediately to prevent recursive mappings
+      vim.keymap.del(mode, key)
+      loader()
       if mode == 'n' then
         vim.api.nvim_input(key)
       else
-        vim.api.nvim_feedkeys(key, mode, false)
+        vim.api.nvim_feedkeys(key, '', false)
```

Comment:
What is this for?

Reply:
`:h feedkeys()` can see:
```vimhelp
		{mode} is a String, which can contain these character flags:
		'm'	Remap keys. This is default.  If {mode} is absent,
			keys are remapped.
		'n'	Do not remap keys.
		't'	Handle keys as if typed; otherwise they are handled as
			if coming from a mapping.  This matters for undo,
			opening folds, etc.
...
```

and `:h vim.keymap.set()`
```vimhelp
      • {mode}  (`string|string[]`) Mode short-name, see |nvim_set_keymap()|.
                Can also be list of modes to create mapping on multiple modes.
```

So this `mode` var cannot be used for feedkeys() semantically.
I saw `lazy.vim` using `'i'` as the value for `mode` for feedkeys()
https://github.com/folke/lazy.nvim/blob/main/lua/lazy/core/handler/keys.lua#L140